.github/workflows/android.yml

```diff
@@ -0,0 +1,134 @@
+name: Dart SDK Android Build
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "dart-sdk@*"
+    paths:
+      - "dart-sdk/rust/**"
+      - "eppo_core/**"
+      - ".github/workflows/android.yml"
+  pull_request:
+    paths:
+      - "dart-sdk/rust/**"
+      - "eppo_core/**"
+      - ".github/workflows/android.yml"
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+env:
+  CI: true
+
+jobs:
+  build-android:
+    name: Build Android (${{ matrix.target }})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          [aarch64-linux-android, x86_64-linux-android, armv7-linux-androideabi]
+        include:
+          - target: aarch64-linux-android
+            android_abi: arm64-v8a
+          - target: x86_64-linux-android
+            android_abi: x86_64
+          - target: armv7-linux-androideabi
+            android_abi: armeabi-v7a
+          - target: i686-linux-android
+            android_abi: x86
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - uses: dart-lang/setup-dart@v1
+        with:
+          sdk: 3.6
+
+      - name: Setup Rust
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Install cross
+        run: cargo install cross --git https://github.com/cross-rs/cross
+
+      - name: Cache Rust dependencies
+        uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: dart-sdk/rust -> target
+          key: ${{ matrix.target }}
+
+      - name: Override eppo_core for testing
+        shell: bash
+        run: |
+          mkdir -p "${CARGO_HOME:-$HOME/.cargo}"
+          echo "[patch.crates-io.eppo_core]" >> "${CARGO_HOME:-$HOME/.cargo}/config.toml"
+          echo "path = '${{ github.workspace }}/eppo_core'" >> "${CARGO_HOME:-$HOME/.cargo}/config.toml"
```

Comment:
does this need to be removed or later after publishing?

Reply:
yeah I think for CI here we want to build against the current unreleased version of eppo_core, but for publishing (assuming that will be in a separate github workflow) we'll want to use the released version instead. So yeah I think this is fine as is, cc @rasendubi to keep me honest

Reply:
oh, yeah. good catch that we need to remove publish against released version of eppo_core

I've updated CI here: https://github.com/Eppo-exp/eppo-multiplatform/pull/229